src/acl/external/file_userip/ext_file_userip_acl.cc

```diff
@@ -61,6 +61,16 @@ int dict_lookup(struct ip_user_dict *, char *, char *);
 /** Size of lines read from the dictionary file */
 #define DICT_BUFFER_SIZE    8196
 
+static void
+free_dict(struct ip_user_dict *head) {
+    while (head) {
+        struct ip_user_dict *next = head->next_entry;
+        free(head->username);
+        free(head);
```

Comment:
Not sure where the request I made for this change disappeared. Anyhow, here it is again.
These are required for UB protection and to go with the `xcalloc` calls.
```suggestion
        safe_free(head->username);
        xfree(head);
```